### ISSUE TITLE
`m.toGams()`: error due to missing parameter in resulting model

### ISSUE THREAD

**Issue by grecht** [GAMS-dev/gamspy#12]
Hi, I encountered an issue where a parameter is missing from a GAMS model that was created using `m.toGams()`. This leads to an error in the GAMS model despite the GAMSPy model running perfectly.

See the example and the resulting GAMS file below. It only occurs if the term `p[S] * x[S]` is squared, so the suspicion is that the squaring has an effect.

The example:
```python
import sys
import gamspy as gp

ct = gp.Container()
S = ct.addSet("S", records=["a"])
p = ct.addParameter("p", domain=S)
p[S] = 2
x = ct.addVariable("x", domain=S, type="positive")

e = ct.addEquation("e", domain=S)

# does not work, as 'p' vanishes
e[S] = (p[S] * x[S]) ** 2 <= 4

# works
# e[S] = p[S] * x[S] <= 4

m = ct.addModel(
    "m",
    gp.Problem.QCP,
    equations=ct.getEquations(),
    sense=gp.Sense.MIN,
    objective=gp.Sum(S, x[S])
)

# solves without problem
m.solve(solver="CPLEX", output=sys.stdout)

# m.gms does not contain parameter 'p'
m.toGams(".")
```
The resulting GAMS file (note that `p` is not declared, but appears in the equation):
```
Set S(*);
positive Variable x(S);
Equation e(S);
free Variable m_objective_variable;
Equation m_objective;
Model m / e,m_objective /;
$gdxLoadAll C:\Users\rech_ge\Documents\workspace\metis\m_data.gdx
e(S) .. ( power((p(S) * x(S)),2) ) =l= 4;
m_objective .. sum(S,x(S)) =e= m_objective_variable;
solve m using QCP MIN m_objective_variable;
```

**Comment by mabdullahsoyturk** [comment]
It is indeed related to the squaring. Thanks for reporting! This will be fixed with the new release.

**Comment by mabdullahsoyturk** [comment]
This was fixed with https://github.com/GAMS-dev/gamspy/commit/adebc20d943198bb2dd1bb70806ac8395246a052. It will be included in 1.0.4.
